trin-core/src/portalnet/overlay.rs

```diff
@@ -376,6 +376,14 @@ where
                 .map(|node| node.value.enr())
                 .collect();
 
+            // Sort all eligible nodes by proximity to the content and take the closest k.
+            interested_enrs.sort_by(|a, b| {
+                let distance_a = XorMetric::distance(&content_key.content_id(), &a.node_id().raw());
+                let distance_b = XorMetric::distance(&content_key.content_id(), &b.node_id().raw());
+                distance_a.partial_cmp(&distance_b).unwrap()
+            });
+            let interested_enrs = interested_enrs[0..MAX_NODES_PER_BUCKET].to_vec();
```

Comment:
What is the reasoning for limiting the interested nodes to 16?

Reply:
@ogenev I updated this variable name to be `closest_k_enrs` so hopefully it's clearer now. 

The reasoning for this limit is that because once we've sorted the list, the first 16 will be the closest to the data. Without this limit, the majority of our gossip will be to nodes whose IDs are far away from the content ID. 

This (including the selection of k=16 nodes) is done according to [portal wire spec](https://github.com/ethereum/portal-network-specs/blob/master/portal-wire-protocol.md#neighborhood-gossip). 

Let me know if that does/doesn't answer your question.

Reply:
> This (including the selection of k=16 nodes) is done according to [portal wire spec](https://github.com/ethereum/portal-network-specs/blob/master/portal-wire-protocol.md#neighborhood-gossip).

I don't understand why k=16 in this case. There are 16 keys per bucket but why do we want to gossip to only 16 nodes instead of all interested nodes that are close to the content?

Reply:
@ogenev It's an arbitrary number. But the larger the number, the more we are gossiping to far-away nodes that will either never be asked or only be asked after asking the closer nodes are asked (and thus both wasting storage space and slowing down queries).

Do you have a different number in mind?

Reply:
I see, here is how Fluffy implements this:

https://github.com/status-im/nimbus-eth1/blob/a74bbef74c49676c8ac78ef4c4a1c8d94265894e/fluffy/network/wire/portal_protocol.nim#L1153

I think their algorithm is an improvement over ours but I guess we can iron it out more in another PR.

Reply:
@ogenev Yea, in a future PR we'll implement the case where there's too few interested nodes so we initiate a `FINDNODES` query.

Currently the max number of nodes we're gossiping to is 4, I'm good with doubling it to 8. You're right, 4 might be too low.

Reply:
8 looks also better to me

> Currently the max number of nodes we're gossiping to is 4, I'm good with doubling it to 8. You're right, 4 might be too low.

Could you factor it into this PR?